test_scripts/AppIconsFolder/ATF_N_AppIconsAmountToRemove_larger_than_icons_amount.lua

```diff
@@ -0,0 +1,326 @@
+require('user_modules/AppTypes')
+
+--[[ Local variables ]]
+local pathToAppFolder
+local file
+local fileContent
+local status = true
+local fileContentUpdated
+local SDLini = config.pathToSDL .. tostring("smartDeviceLink.ini")
+local RAIParameters = config.application1.registerAppInterfaceParams
+local applicationFileToCheck = config.pathToSDL .. tostring("Icons/" .. RAIParameters.appID)
+local firstOldFileToCheck = config.pathToSDL .. tostring("Icons/icon1.png")
+local secondOldFileToCheck = config.pathToSDL .. tostring("Icons/icon2.png")
+local thirdOldFileToCheck = config.pathToSDL .. tostring("Icons/icon3.png")
+
+--[[ Local functions ]]
+local function registerApplication(self)
+  local corIdRAI = self.mobileSession:SendRPC("RegisterAppInterface", RAIParameters)
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+  {
+    application =
+    {
+     appName = RAIParameters.appName
+    }
+  })
+  :Do(function(_,data)
+    self.applications[RAIParameters.appName] = data.params.application.appID
+  end)
+  self.mobileSession:ExpectResponse(corIdRAI, { success = true, resultCode = "SUCCESS" })
+end
+
+-- Check file existence 
+local function checkFileExistence(name, messages)
+  file=io.open(name,"r")
+  if file ~= nil then
+    io.close(file)
+    if messages == true then
+      commonFunctions:userPrint(32, "File " .. tostring(name) .. " exists")
+    end
+    return true
+  else
+    if messages == true then
+      commonFunctions:userPrint(31, "File " .. tostring(name) .. " does not exist")
+    end
+    return false
+  end
+end
+
+--Check path to SDL in case last symbol is not'/' add '/'
+local function checkSDLPathValue()
+  local findResult = string.find (config.pathToSDL, '.$')
+  if string.sub(config.pathToSDL,findResult) ~= "/" then
+    config.pathToSDL = config.pathToSDL..tostring("/")
+  end
+end
+
+-- Generate path to application folder
+local function pathToAppFolderFunction(appID)
+  checkSDLPathValue()
+  local path = config.pathToSDL .. tostring("storage/") .. tostring(appID) .. "_" .. tostring(config.deviceMAC) .. "/"
+  return path
+end
+
+-- Check directory existence
+local function checkDirectoryExistence(DirectoryPath)
+  local returnValue
+  local command = assert( io.popen(  "[ -d " .. tostring(DirectoryPath) .. " ] && echo \"Exist\" || echo \"NotExist\"" , 'r'))
+  os.execute("sleep 0.5")
+  local commandResult = tostring(command:read( '*l' ))
+    if commandResult == "NotExist" then
+      returnValue = false
+    elseif
+      commandResult == "Exist" then
+      returnValue =  true
+    else
+      commonFunctions:userPrint(31," Unexpected result in checkDirectoryExistence function, commandResult = " .. tostring(commandResult))
+      returnValue = false
+    end
+    return returnValue
+end
+
+-- Get folder size
+local function dSize(PathToFolder)
+  local sizeFolderInBytes
+  local aHandle = assert( io.popen( "du -sh " ..  tostring(PathToFolder), 'r'))
```

Comment:
@aarotar, Disagree. Folder size in bytes can be easily calculated by executing 'du -s -B1'
In this way folderSize() local function will be much simplier